importccl: support DELETE FROM geometry_columns / geography_columns

otan opened the pull request:
The functionality to delete from geometry_columns / geography_columns is
is not supported in CRDB. As such, when importing these files from
ogr2ogr dumps using IMPORT PGDUMP, ignore the DELETE statement.

This has crept up because of stricter checking from IMPORT PGDUMP.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53110)
<!-- Reviewable:end -->

dt commented on pkg/ccl/importccl/read_import_pgdump.go:
```diff
@@ -488,6 +488,17 @@ func readPostgresStmt(
 		// ignore txns.
 	case *tree.SetVar, *tree.Insert, *tree.CopyFrom, copyData:
 		// ignore SETs and DMLs.
+	case *tree.Delete:
```
Nit: I think I’d put `*tree.Delete` in the case above this to just fast-ignore it along with other DMLs,  since here is during schema extraction. Instead I’d move what you have in this case to a stand-alone `Delete` case in the data extraction pass below (where you have it unconditionally ignored)

otan replied:
Done! Cheers for the late night peek.

otan commented:
bors r=dt

otan commented:
bors r-

otan committed:
importccl: support DELETE FROM geometry_columns / geography_columns

The functionality to delete from geometry_columns / geography_columns is
is not supported in CRDB. As such, when importing these files from
ogr2ogr dumps using IMPORT PGDUMP, ignore the DELETE statement.

This has crept up because of stricter checking from IMPORT PGDUMP.

Release note: None

otan commented:
forgot to commit the update, woops

bors r=dt

otan commented:
bors r-

otan commented:
bors r=dt
